### PR TITLE
Add err code to prometheus metrics

### PR DESCRIPTION
**Pull request by unkeep**

### CODE REVIEW

File: prometheus/prometheus.go [hunk 1]
```diff
@@ -2,17 +2,22 @@ package prometheus
...
 	prometheusclient "github.com/prometheus/client_golang/prometheus"
+	"github.com/vektah/gqlparser/gqlerror"
 )
 
 const (
 	existStatusFailure = "failure"
 	exitStatusSuccess  = "success"
 )
 
+const errCodeExt = "error_code"
```
Comment (mkorolyov) [inline prometheus/prometheus.go]: join const declaration with 2 already existing above?

**Review by mkorolyov — CHANGES_REQUESTED** [review]

File: prometheus/prometheus.go (hunk 1 above)
Comment (unkeep) [reply prometheus/prometheus.go]: I prefer separate independent const/var groups

Comment (mkorolyov) [reply prometheus/prometheus.go]: but they are connected. 3 of them for a single Prometheus metric.

File: prometheus/prometheus.go [hunk 2]
```diff
@@ -109,7 +113,17 @@ func ResolverMiddleware() graphql.FieldMiddleware {
 			exitStatus = exitStatusSuccess
 		}
 
-		timeToResolveField.WithLabelValues(exitStatus, rctx.Object, rctx.Field.Name).
+		var errCode string
+		if err != nil {
+			gqlErr := &gqlerror.Error{}
```
Comment (auvn) [inline prometheus/prometheus.go]: nit: `var gqlErr *gqlerror.Error`

Comment (unkeep) [reply prometheus/prometheus.go]: >  second argument to errors.As must be a non-nil pointer to either a type that implements error, or to any interface type
code from gqlerror pkg, error.go:
```go
func (err *Error) Error() string {...}
```
only ptr on `Error` implement std error

Comment (auvn) [reply prometheus/prometheus.go]:
```
var gqlErr *gqlerror.Error
errors.As(err, &gqlErr)
```

is a non nil pointer. It's a pointer of **gqlerror.Error type to nil value of *gqlerror.Error which implements error interface

https://play.golang.org/p/foQs4b9c2L5

Comment (auvn) [reply prometheus/prometheus.go]: the main idea of this: you don't need to allocate a whole type value if error type is different

Comment (unkeep) [reply prometheus/prometheus.go]: yep, u are right!

Comment (unkeep) [reply prometheus/prometheus.go]: fixed

**Comment by auvn** [comment]
I'm ok. but cannot approve :(

**Pull request merged**